core/src/main/scala/org/broadinstitute/workbench/hamm/pricing/GcpPricing.scala

```diff
@@ -1,30 +1,95 @@
+      val sku = googlePriceList.priceItems.filter { priceItem =>
+        (priceItem.regions.contains(region)
+          && priceItem.category.resourceFamily.equals(resourceFamily)
+          && priceItem.category.resourceGroup.equals(resourceGroup)
+          && priceItem.category.usageType.equals(usageType)
+          && (descriptionShouldInclude match {
+          case Some(desc) => priceItem.description.asString.contains(desc)
+          case None => true})
+          && (descriptionShouldNotInclude match {
+          case Some(desc) => !priceItem.description.asString.contains(desc)
+          case None => true}))
+      }
+      sku.length match {
+        case 0  => Left(s"No SKUs matched with region $region, resourceFamily $resourceFamily, resourceGroup $resourceGroup, $usageType usageType, and description including $descriptionShouldInclude and notIncluding $descriptionShouldNotInclude in the following price list: ${googlePriceList.priceItems}")
+        case 1 => getPriceFromSku(sku.head)
+        case tooMany => Left(s"$tooMany SKUs matched with region $region, resourceFamily $resourceFamily, resourceGroup $resourceGroup, $usageType usageType, and description including $descriptionShouldInclude and notIncluding $descriptionShouldNotInclude. ${sku.toString} were in the following price list: ${googlePriceList.priceItems}")
+      }
+    }
+
+    def getPriceFromSku(priceItem: GooglePriceItem): Either[String, Double] = {
+      // ToDo: Currently just takes first, make it take either most recent or make it dependent on when the call ran
+      priceItem.pricingInfo.headOption match {
+        case None => Left(s"Price Item $priceItem had no pricing info")
+        case Some(head) => Right(head.tieredRates.filter(rate => rate.startUsageAmount.asInt == 0).head.nanos.asInt.toDouble / 1000000000)
+      }
+    }
+
+    def getComputePrices(computePriceKey: ComputePriceKey): Either[String, ComputePrices] = {
+      for {
+        cpuPrice <- getPrice(computePriceKey.region, ResourceFamily.Compute, ResourceGroup(computePriceKey.machineType.asCPUresourceGroupString), computePriceKey.usageType, Some(computePriceKey.machineType.asDescriptionString), None)
+        ramPrice <- getPrice(computePriceKey.region, ResourceFamily.Compute, ResourceGroup(computePriceKey.machineType.asRAMresourceGroupString), computePriceKey.usageType, Some(computePriceKey.machineType.asDescriptionString), None)
+      }  yield ComputePrices(cpuPrice, ramPrice)
+    }
+
+    def getStoragePrice(storagePriceKey: StoragePriceKey): Either[String, Double] = {
+        getPrice(storagePriceKey.region, ResourceFamily.Storage, ResourceGroup(storagePriceKey.diskType.asString), UsageType.OnDemand, None, Some("Regional"))
+    }
+
+    val computePrices: Seq[Either[String, (ComputePriceKey, ComputePrices)]] = computePriceKeys.map{ key =>
```

Comment:
Any reason for `val` vs `def` here?

Reply:
I think this should stay a val but the storagePrices right below it should be changed to a val as well. thanks :)